### ISSUE TITLE
Rule 27 incorrectly fails on '*' wildcard with USING clauses

### ISSUE THREAD

**Issue by epgui** [sqlfluff/sqlfluff#4366]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

SQLFluff 1.3.2 will fail the following:

```
SELECT *
FROM table_a
INNER JOIN table_b USING (column_a, column_b)
INNER JOIN table_c USING (column_a, column_b)
INNER JOIN table_d USING (column_a, column_b)
INNER JOIN table_e USING (column_a, column_b)
```

```bash
L:  46 | P:  12 | L027 | Unqualified reference '*' found in select with more than
                       | one referenced table/view.
```

However, this works as expected:

```
SELECT
    column_a,
    column_b
FROM table_a
INNER JOIN table_b USING (column_a, column_b)
INNER JOIN table_c USING (column_a, column_b)
INNER JOIN table_d USING (column_a, column_b)
INNER JOIN table_e USING (column_a, column_b)
```

Rule 027 notes: "Except if they’re present in a USING clause." but makes no mention of wildcards not being allowed.

### Expected Behaviour

Either SQLFluff should not fail on the provided input, or the docs should indicate that this is expected behaviour.

### Observed Behaviour

See above.

### How to reproduce

See above.

### Dialect

PostgreSQL

### Version

sqlfluff, version 1.3.2
python version 3.10.8

### Configuration

```
[sqlfluff]
dialect = postgres
; L032: Prefer specifying join keys instead of using USING.
; This rule was originally intended to encourage better intercompatibility
; across databases, but it would appear that whatever issues existed with USING
; statements no longer exist, and that some would now encourage their use.
; See also:
;   - https://docs.sqlfluff.com/en/stable/rules.html#sqlfluff.rules.Rule_L032
;   - https://github.com/dbt-labs/corp/pull/58
exclude_rules = L032
templater = jinja

[sqlfluff:rules]
aliasing = explicit
comma_style = trailing
indent_unit = space
max_line_length = 100
tab_space_size = 4

; L003: Indentation not consistent with previous lines.
[sqlfluff:rules:L003]
hanging_indents = false

; L007: Operators should follow a standard for being before/after newlines.
[sqlfluff:rules:L007]
operator_new_lines = before

; L010: Inconsistent capitalisation of keywords.
[sqlfluff:rules:L010]
capitalisation_policy = upper

; L014: Inconsistent capitalisation of unquoted identifiers.
[sqlfluff:rules:L014]
extended_capitalisation_policy = lower
unquoted_identifiers_policy = all

; L016: Line is too long.
[sqlfluff:rules:L016]
ignore_comment_clauses = false
ignore_comment_lines = false

; L029: Keywords should not be used as identifiers.
[sqlfluff:rules:L029]
quoted_identifiers_policy = none
; This setting should probably be set, but I'm not sure it works properly:
; unquoted_identifiers_policy = all

; L030: Inconsistent capitalisation of function names.
[sqlfluff:rules:L030]
extended_capitalisation_policy = upper

; L040: Inconsistent capitalisation of boolean/null literal.
[sqlfluff:rules:L040]
capitalisation_policy = upper

; L042: Join/From clauses should not contain subqueries. Use CTEs instead.
[sqlfluff:rules:L042]
forbid_subquery_in = both

; L054: Inconsistent column references in 'GROUP BY/ORDER BY' clauses.
[sqlfluff:rules:L054]
group_by_and_order_by_style = explicit

; L057: Do not use special characters in identifiers.
[sqlfluff:rules:L057]
allow_space_in_identifier = false
quoted_identifiers_policy = all
unquoted_identifiers_policy = all

; L063: Inconsistent capitalisation of datatypes.
[sqlfluff:rules:L063]
extended_capitalisation_policy = upper

; L064: Consistent usage of preferred quotes for quoted literals.
[sqlfluff:rules:L064]
preferred_quoted_literal_style = single_quotes

[sqlfluff:templater:jinja]
apply_dbt_builtins = true
library_path = sqlfluff_libs
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by epgui** [comment]
Apologies, I believe I may have not fully understood how this was supposed to work.

For anyone else coming up against this, columns not included in the USING clause are apparently not exempt from the rule.
